src/plugins/kibana_utils/common/persistable_state/index.ts

```diff
@@ -21,9 +23,33 @@ export type MigrateFunction<
 > = (state: FromVersion) => ToVersion;
 
 export type MigrateFunctionsObject = {
-  [key: string]: MigrateFunction;
+  [key: string]: MigrateFunction<any, any>;
 };
 
+export const mergeSavedObjectMigrationMaps = (
+  obj1: SavedObjectMigrationMap,
+  obj2: SavedObjectMigrationMap
+) => {
+  const customizer = (objValue: MigrateFunction, srcValue: MigrateFunction) => {
+    if (!srcValue || !objValue) {
+      return srcValue || objValue;
+    }
+    return (state: SerializableState) => objValue(srcValue(state));
+  };
+
+  return mergeWith(obj1, obj2, customizer);
```

Comment:
Maybe it is better to write this function without using `lodash`, as for something that simple feels wrong to include `lodash` library.

Reply:
I'm not 100% sure, but I guess `mergeWith` mutates the first argument `obj1`. Or we OK with that? Or better to leave it immutable?

```suggestion
  return mergeWith({ ...obj1 }, obj2, customizer);
```